simcon_project/templates/create_assignment.html

```diff
@@ -0,0 +1,312 @@
+                  var tempKey=name+"--"+em;
+                  var index = addedTmp.indexOf(tempKey);
+                  if (index > -1) {
+                      addedTmp.splice(index, 1);
+                  }
+              }
+              //label
+              else{
+                  var index = addedLabel.indexOf(em);
+                  if (index > -1) {
+                      addedLabel.splice(index, 1);
+                  }
+              }
+          };
+          //Add cells and rows to the table
+          tr.appendChild(tUsername);
+          tr.appendChild(tKey);
+          tr.appendChild(tDel);
+          tBody.appendChild(tr);
+      }
+
+      function add_stu(tbl, stu){
+          var find=false;   //Record find the corresponding student or not
+          for(var i=0;i<students.length;i++){
+              var stuTmp=students[i];
+              if(stuTmp.email==stu){
+                  find=true;
+                  if(stuTmp.is_active!=1){
+                      window.alert("Add failed. This user is a not an active user");
+                  }
+                  else{
+                      //Determine whether this student is already in the table.
+                      if(addedStu.indexOf(stu)>-1){
+                          window.alert("Adding failed, the student is already in the table.");
+                      }
+                      else {
+                          var name = stuTmp.first_name + " " + stuTmp.last_name;
+                          add2table(tbl, name, stu, 1);
+                          addedStu.push(stu);
+                      }
+                  }
+              }
+          }
+          if(find==false){
+              window.alert("Cannot find: "+stu);
+          }
+          document.getElementById('stu_name').value="";
+      }
+
+      function add_temp(tbl, name){
+          var find=false;
+          for(var i=0;i<templates.length;i++){
+              var tempTmp=templates[i].fields;
+              if(tempTmp.name==name){
+                  find=true;
+                  var key=name+"--"+tempTmp.creation_date
+                  if(addedTmp.indexOf(key)>-1){
+                      window.alert("Adding failed, the template is already in the table.");
+                  }
+                  else {
+                      add2table(tbl1, name, tempTmp.creation_date, 2);
+                      addedTmp.push(key);
+                  }
+              }
+          }
+          if(find==false){
+              window.alert("Cannot find: "+name);
+          }
+          document.getElementById('template_name').value="";
+      }
+
+      function add_label(tbl, name){
+          var find=false;
+          for(var i=0;i<subLabels.length;i++){
+              var labelTmp=subLabels[i].fields;
+              if(labelTmp.label_name==name){
+                  find=true;
+                  if(addedLabel.indexOf(name)>-1){
+                      window.alert("Adding failed, the label is already in the table.");
```

Comment:
I dont know if we should error, or ignore and keep going

Reply:
I think to be consistent with the rest, we should go with popping up an error.